Adding NBA_elo & NBA_elo_latest datasets

kvanallen opened the pull request:
I added the nba_elo and nba_elo_latest datasets and accompanying documentation to the FiveThirtyEight package.

Commit:
adding data sets

Commit:
last commit

Commit:
small space

Commit:
updating fork

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NEWS.md

beanumber commented on R/process_data_sets_kara.R:
```diff
@@ -0,0 +1,26 @@
+#setting up .rda
+usethis::use_data(nba_elo, overwrite = TRUE)
+usethis::use_data(nba_elo_latest, overwrite = TRUE)
+
```
@kvanallen you want this file to be in the `data-raw/` folder, not `R`. Can you please move it?

kvanallen replied:
fixed! thank you!

beanumber commented on R/process_data_sets_kara.R:
```diff
@@ -0,0 +1,26 @@
+#tidying and taming data for nba_elo
+nba_elo %>% clean_names()
+gsub(x = colnames(nba_elo), pattern = "-", replacement = "_")
+nba_elo[13:22] <- as.numeric(as.character(nba_elo[13:22]))
```
Not sure how @rudeboybert will feel about this use of hard-coded indices. I guess it's OK since these files are static. It would be more expressive to use `mutate_at()` or something like that.

kvanallen replied:
thank you! I changed it to a mutate_if statement.

rudeboybert replied:
Yep, never hardcode anything. thanks

Commit:
changing file location to data-raw

Commit:
changing location to data-raw file

Commit:
recoding columns using mutate_if

rudeboybert commented:
Tagging #53

beanumber commented:
From Travis:
```
Error in data.frame(package = deps, installed = inst_ver, available = cran_ver,  : 

  row names contain missing values
```
@kvanallen are there row names? If so, can you remove them?

beanumber commented:
@kvanallen can you make these changes? You also may need to [sync your fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork)

Commit:
eliminating row names

Commit:
merge upstream

Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	DESCRIPTION
#	NEWS.md

rudeboybert committed:
Merge branch 'master' into master

rudeboybert commented on DESCRIPTION:
```diff
@@ -29,6 +29,7 @@ Authors@R: c(
     person("Sunni", "Raleigh", email = "sunnigraleigh@gmail.com", role = "ctb"),
     person("Anna", "Ballou", email = "afballou16@gmail.com", role = "ctb"),
     person("Jane", "Bang", email = "carolbjy@gmail.com", role = "ctb"),
+    person("Kara", "Van Allen", email = "kkv354@gmail.com" role = "ctb"),
```
You are missing a space after your email, thus causing the package to not even be able to build.

rudeboybert replied:
Sorry, you're missing a comma.

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,31 @@
+rownames(nba_elo_latest) <- NULL
+
+#tidying and taming data for nba_elo
+nba_elo %>% clean_names()
```
you pipe `nba_elo` here into to `clean_names()`, but don't save it. As a resulting we have variable names that have the ugly tick marks. For example `carm-elo1_pre` should be `carm_elo1_pre`
```
> glimpse(nba_elo)
Observations: 69,636
Variables: 24
$ date             <date> 1946-11-01, 1946-11-02, 1946-11-02, 1946-11-02, 1946-11-…
$ season           <int> 1947, 1947, 1947, 1947, 1947, 1947, 1947, 1947, 1947, 194…
$ neutral          <int> 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, …
$ playoff          <chr> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ team1            <chr> "TRH", "CHS", "PRO", "STB", "DTF", "CLR", "PIT", "BOS", "…
$ team2            <chr> "NYK", "NYK", "BOS", "PIT", "WSC", "TRH", "WSC", "CHS", "…
$ elo1_pre         <dbl> 1300.000, 1300.000, 1300.000, 1300.000, 1300.000, 1300.00…
$ elo2_pre         <dbl> 1300.000, 1306.723, 1300.000, 1300.000, 1300.000, 1293.27…
$ elo_prob1        <dbl> 0.6400650, 0.6311012, 0.6400650, 0.6400650, 0.6400650, 0.…
$ elo_prob2        <dbl> 0.3599350, 0.3688988, 0.3599350, 0.3599350, 0.3599350, 0.…
$ elo1_post        <dbl> 1293.277, 1309.652, 1305.154, 1304.691, 1279.619, 1307.12…
$ elo2_post        <dbl> 1306.723, 1297.071, 1294.846, 1295.309, 1320.381, 1286.15…
$ `carm-elo1_pre`  <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ `carm-elo2_pre`  <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ `carm-elo_prob1` <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ `carm-elo_prob2` <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ `carm-elo1_post` <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ `carm-elo2_post` <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ raptor1_pre      <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ raptor2_pre      <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ raptor_prob1     <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ raptor_prob2     <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, N…
$ score1           <int> 66, 63, 59, 56, 33, 71, 56, 55, 49, 81, 73, 63, 73, 76, 7…
$ score2           <int> 68, 47, 53, 51, 50, 60, 71, 57, 53, 75, 65, 68, 71, 66, 6…
```

rudeboybert replied:
Could you also convert both `team` variables to factors?

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,31 @@
+
+#tidying and taming data for nba_elo_latest
+nba_elo_latest %>% clean_names()
+gsub(x = colnames(nba_elo_latest), pattern = "-", replacement = "_")
```
Same as above.

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,31 @@
+
+#tidying and taming data for nba_elo
+nba_elo %>% clean_names()
+gsub(x = colnames(nba_elo), pattern = "-", replacement = "_")
```
If this ends up being necessary after the `clean_names()`, could you use `str_replace_all()` inside of a `mutate()` in the `%>%` chain instead?

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,31 @@
+nba_elo %>% clean_names()
+gsub(x = colnames(nba_elo), pattern = "-", replacement = "_")
+
+nba_elo <- nba_elo %>%
```
For `neutral` and `playoff`, could you make them `TRUE/FALSE` logical and update the help file accordingly?

rudeboybert replied:
Still outstanding

rudeboybert commented on R/data_kara.R:
```diff
@@ -0,0 +1,49 @@
+#' NBA Elo Ratings Through History
```
Please format this line so that it matches the title of the article

rudeboybert commented on R/data_kara.R:
```diff
@@ -0,0 +1,49 @@
+#' NBA Elo Ratings Through History
+#' 
+#' The data behind the Elo and RAPTOR ratings for NBA teams used in various stories like this one: 
```
We have been using a consistent standard for the description of all datasets, could you please make this match too? Here is an example: https://github.com/rudeboybert/fivethirtyeight/blob/master/R/data_anna.R#L3

rudeboybert replied:
Could you make all the fixes needed in this help file?

rudeboybert commented on R/data_kara.R:
```diff
@@ -0,0 +1,49 @@
+#' NBA Elo Ratings Through History
+#' 
+#' The data behind the Elo and RAPTOR ratings for NBA teams used in various stories like this one: 
+#' \url{https://projects.fivethirtyeight.com/complete-history-of-the-nba/#raptors}
```
Remove the `#raptors` so that menu starts blank

rudeboybert commented on R/data_kara.R:
```diff
@@ -0,0 +1,49 @@
+#' The data behind the Elo and RAPTOR ratings for NBA teams used in various stories like this one: 
+#' \url{https://projects.fivethirtyeight.com/complete-history-of-the-nba/#raptors}
+#' 
+#'   @format A data frame with 69636 rows representing every game in the NBA, and 24 variables:
```
The extra spacing in front of @format is causing the help file to come out wonky. Could you fix and make sure to read the resulting help file?

rudeboybert commented on R/data_kara.R:
```diff
@@ -0,0 +1,49 @@
+#'   
+"nba_elo"
+
+#' NBA Elo Ratings for this season
```
Also, since these two data frames are very related, it makes sense to combine them into the same help file. See this example: https://github.com/rudeboybert/fivethirtyeight/blob/master/R/data_marium.R

rudeboybert reviewed (commented):
Hey @kvanallen, thanks for your PR. I've wrote out necessary changes. Please ensure that before your commits you:
1. Ensure the package can actually build
1. Run the package checks so that obvious errors get caught

Commit:
fixed suggested in data-raw

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,35 @@
+nba_elo_latest <- nba_elo_latest %>%
+  mutate_if(is.logical, as.numeric)
+
+nba_elo_latest$neutral <- as.logical(nba_elo_latest$neutral)
```
Could you use:
* `mutate()` instead?
* Combine all `%>%` into a single pipe chain? 

Kind of like in this section of the [tidyverse style guide](https://style.tidyverse.org/pipes.html#whitespace)

rudeboybert reviewed (commented):
Hey Kara, thanks for your update. A few more fixes are required to get this data merged into the package. On top of some of the still outstanding changes in the various files, could you:

* Build the package and ensure both the help file and the data look good
* Check the package so that you're alerted to package compilation errors
* The push/commit the changes? This PR will update itself.

thanks. tagging @beanumber

Commit:
changing name

Commit:
tidy data

beanumber commented:
I think this Travis failure can be ignored. It doesn't seem to have anything to do with this PR. @rudeboybert

rudeboybert commented:
Yep, you can ignore the travis failure. But the package checks say that there is a mismatch between the `roxygen2` help file code and the variable names in the data frame. Also, a few remaining data wrangling clean-ups in `process_data_sets_kara.R` and then we'll be ready to merge

Commit:
combine help files as suggested

Commit:
make all logical coded as such

kvanallen commented:
I combined the help files and cleaned up the data wrangling. Please let me know if more needs to be done. Thank you so much!

kvanallen committed:
Merge branch 'master' into master

rudeboybert commented on DESCRIPTION:
```diff
@@ -29,8 +29,10 @@ Authors@R: c(
+    person("Kara", "Van Allen", email = "kkv354@gmail.com", role = "ctb"),
     person("Jessica", "Keast", email = "jeskeast@gmail.com", role = "ctb"),
     person("Fatima", "Keita", email = "fkeita898@gmail.com", role = "ctb"))
+
```
You'll need to remove this blank line. Having it causes the package build to yield an error.

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,35 @@
+#tidying and taming data for nba_elo
+nba_elo <- nba_elo %>% clean_names()
+
+nba_elo <- nba_elo %>%
```
Could you combine lines 15 thru 20 into a single pipe chain, with one operation per line?

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,35 @@
+
+
+#tidying and taming data for nba_elo_latest
+nba_elo_latest <- nba_elo_latest %>% clean_names()
```
Could you combine lines 24 thru 29 into a single pipe chain, with one operation per line?

rudeboybert commented on data-raw/process_data_sets_kara.R:
```diff
@@ -0,0 +1,35 @@
+
+#setting up .rda
+usethis::use_data(nba_elo, overwrite = TRUE)
+usethis::use_data(nba_elo_latest, overwrite = TRUE)
```
After you've made the above changes, could you re-run this code in its entirety? The `data/nba_elo.rda` and `data/nba_elo_latest.rda` data objects appear to be out of date, hence need to be rebuilt and then committed and pushed as part of this pull request.

rudeboybert reviewed (commented):
Hey Kara, thanks for your updates. You're almost done! Just a few more changes to make, then build/check the package, then commit/push. We should be able to merge your pull request after these changes are successfully made!

Commit:
final changes

kvanallen commented:
Thank you! I fixed the final changes.

rudeboybert commented:
Congrats! Your PR is merged! you're now listed as a one of the package [contributors](https://github.com/rudeboybert/fivethirtyeight/graphs/contributors)
